stream_test.py

```diff
@@ -0,0 +1,20 @@
+import unittest
+import generate_stream
+
+class GenerateStreamTest(unittest.TestCase):
+  def test_within_range_temp(self):
+    TestReading=generate_stream.generate_param_reading('temperature')
+    self.assertTrue((TestReading>=generate_stream.Battery_parameters_range['temperature'][0]) and (TestReading<=generate_stream.Battery_parameters_range['temperature'][1]))
+  def test_within_range_soc(self):
+    TestReading=generate_stream.generate_param_reading('temperature')
+    self.assertTrue((TestReading>=generate_stream.Battery_parameters_range['temperature'][0]) and (TestReading<=generate_stream.Battery_parameters_range['soc'][1]))
+  def test_number_of_readings_generated(self):
```

Comment:
No tests for the output format... that is the crucial functionality of the Sender.